sklearn/utils/testing.py

```diff
@@ -647,8 +648,20 @@ def is_abstract(c):
     return sorted(set(estimators))
 
 
+NON_RANDOM_CLASSES = (DBSCAN,)
+
+
 def set_random_state(estimator, random_state=0):
```

Comment:
If there is just one class that it is deprecated for, it might be better to just do

    if isinstance(estimator, DBSCAN):
        return